src/main/java/hello/interceptor/HeaderValidator.java

```diff
@@ -0,0 +1,39 @@
+package hello.interceptor;
+import hello.exception.BadRequestException;
+import org.apache.log4j.BasicConfigurator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler. HandlerInterceptorAdapter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.Optional;
+
+@Component
+public class HeaderValidator extends HandlerInterceptorAdapter {
+
+    static{
+        BasicConfigurator.configure();
+    }
+
+    @Autowired
+    private ValidatorInterceptor[] validatorInterceptors;
```

Comment:
Key point of learning.  Spring component scanning will detect every ValidatorInterceptor implementation and it will be used in the preHandle easily.